### ISSUE TITLE
Clarify the no-body logs

### ISSUE THREAD

**Issue by lmazuel** [Azure/azure-sdk-for-python#12846]
This is the way the network logger is logging the body:
https://github.com/Azure/azure-sdk-for-python/blob/0021b7b0ba4ee9276f6313631373b0e2d6f584a4/sdk/core/azure-core/azure/core/pipeline/policies/_universal.py#L292

This might be confusing, to make an explicit difference between "there was no body, it was None" and "there was a 4 bytes body with the letters N-o-n-e" ([example](https://github.com/Azure/autorest.python/issues/733)).

The latter is actually something that could happen, if any layer is doing an incorrect `str(None)` at some point in the code, and debugging using pure loggers will be complicated.

This issue is about splitting explicitly the `None` body as a separate line, something like:
```python
elif request.body is None:
    _LOGGER.debug("This request has no body")
else:
    _LOGGER.debug(str(http_request.body))
```

**Comment by xiangyan99** [comment]
#12830

**Comment by xiangyan99** [comment]
Fixed in #12830
